stories/components/BadgeState.stories.mdx

```diff
@@ -0,0 +1,77 @@
+
+import { Canvas, Meta, Story, ArgsTable, Source } from '@storybook/addon-docs';
+import BadgeState from '@/pkg/rancher-components/src/components/BadgeState/BadgeState';
+
+<Meta 
+  title="Components/BadgeState" 
+  component={BadgeState}
+/>
+
+export const Template = (args, { argTypes }) => ({
+  components: { BadgeState },
+  props:      Object.keys(argTypes),
+  template:   '<BadgeState v-bind="$props" />',
+})
+
+
+# BadgeState
+
+Badges are used to highlight important changes of state in an element, mark them, or draw attention to counters.
+
+<br/>
+
+### Description
+
+- Use badges to highlight or differentiate key elements. However, do it with caution. Drawing attention to many pieces may result in creating much noise.
+- Badges are good for counters, for example, to show the number of new messages.
+- Badges should not provide interaction. However, its parent element may.
+- Don't use long texts within a badge. Whenever possible, limit it to a maximum of 3 words or less. 
+
+
+Different contextual variations of the badge can be used to communicate the different states of an element or the content: Blue for information, yellow for warning and red for errors.
+
+<Canvas columns={2}>
+  Info
+  <Story
+    name="Info"
+    args={{
+      label: 'Hello world',
+      color: 'bg-info'
+    }}>
+    {Template.bind({})}
+  </Story>
+  Warning
+  <Story
+    name="Warning"
+    args={{
+      label: 'Hello World',
+      color: 'bg-warning'
+    }}>
+    {Template.bind({})}
+  </Story>
+  Error
+  <Story
+    name="Error"
+    args={{
+      label: 'Hello World',
+      color: 'bg-error'
+    }}>
+    {Template.bind({})}
+  </Story>
+</Canvas>
+
+### Import
+
+<Source
+  language='js'
+  light
+  format={false}
+  code={`
+     import BadgeState from '@/pkg/rancher-components/src/components/BadgeState/BadgeState';
+  `}
+/>
+
+### Props table
+
+<ArgsTable of={BadgeState} />
```

Comment:
Can we get information about that what these error states are as well? I see `info`, `error`, `warning` and `success`

So it would be good to have these documented. e.g. if user define

`BadgeState = 'alert'` for e.g. this won't work. So it would be good to see this before drilling down. 

Also an example of when no state is provided. If state need to provided then the it should be declared as a required prop.

Reply:
We have already mentioned information about different states below the Description para.

I did not get why users will define it. `BadgeState = 'alert'`. 

We are showing how to implement code eg

![Screenshot 2022-08-04 at 14 18 06](https://user-images.githubusercontent.com/18264463/182845089-82139fb1-f930-4f75-b196-7f5d22b86c74.png)

Reply:
I should have been more clearer. I meant to say it should have the success badge state as well. So all the possible states are shown.